File: Typescript/TwoSum.ts
export interface IHash {
    [props: number] : number;
}

function twoSum2(nums: number[], target: number): number[] {
    for(let i : number = 0; i<nums.length; i++){
        for(let j: number = i+1; j<nums.length; j++){
            if(nums[i] + nums[j] === target) return [i,j];
        }
    }
};

function twoSum(nums: number[], target: number): number[] {
    let hm: IHash = {};

    for(let i : number = 0; i<nums.length; i++){
        if(hm.hasOwnProperty(target - nums[i])) return [i, hm[  target-nums[i] ]]; 
        else hm[nums[i]] = i;
    }
    return [];
};



console.log(twoSum2([2,7,11,15],9));

File: Go/TwoSum.go
package main

func twoSum(nums []int, target int) []int {
	var idMap map[int]int
	idMap = make(map[int]int)

	for i := 0; i < len(nums); i++ {
		v, found := idMap[target-nums[i]]
		if found {
			return []int{v, i}
		} else {
			idMap[nums[i]] = i
		}
	}

	return []int{}
}
